fix(admin): Remove the admin login

mlissner opened the pull request:
Fixes: #2251

albertisfu commented:
This looks good to me, I only found one thing that maybe is not related. On my dev machine, I created my superuser via Django shell. When creating a super user using Django shell no User Profile is assigned to the superuser. 

So now using the rate-limited login page raises an error: profile user doesn't exist. So it's not possible to login in for super users created via Django shell.

![Screen Shot 2022-08-09 at 11 59 56](https://user-images.githubusercontent.com/486004/183713795-6453d51a-221c-48e3-95ed-f9bb8db283dc.png)
I solved this by adding a profile to my superuser.

Not sure if this might be a problem for developers.

I also saw that mypy is failing, I think it might be related to [2125](https://github.com/freelawproject/courtlistener/pull/2125) because I remember and it's the same error message. I can check it out, if that's ok.

mlissner commented:
Yeah, if you want to check out the mypy error, that'd be great, thank you! I was just annoyed with that after fixing the other CI issue last night. Arg!

mlissner commented:
> So now using the rate-limited login page raises an error: profile user doesn't exist. So it's not possible to login in for super users created via Django shell.

This has bugged me for many years, but I've been too lazy to fix it and just worked around it myself. Now that it's affecting others, I put in the 15 minutes to fix it (let's see if tests pass). 

I also consolidated the other signals and gave them all dispatch_uid fields, just in case. I'm pretty bad at signals, and I think you're more used to using them. I think this all works, but please let me know if you see anything that could be better.

mlissner commented:
Well, OK two days later, it turns out this was a good instinct:

> let's see if tests pass

They did not. And fixing them was really difficult because we didn't have the factories we needed and a lot of tests used a fixture that created profiles in a way that didn't work with the new signal. 

So I did a lot of refactoring. Tests now pass again and I removed four fixtures, one of which was used in a lot of places.

Also:

 - In https://github.com/freelawproject/courtlistener/pull/2252/commits/1cea2246b10cd0217b46e162176c95c02b43b76a, I realized that we weren't counting stats about bulk data downloads, but were still showing them to users, so I removed the user part.
 - In https://github.com/freelawproject/courtlistener/pull/2252/commits/1f67168677145fd51dfef512d6889e21257670bb, I abstracted out the way we create a redis key so that it could be mocked by tests. This was necessary to allow tests to run in parallel. Now that they're faster (no more fixtures), they were conflicting a lot more. One of our tests checks that APIs log properly, and it couldn't get a good count in redis because other API tests were logging at the same time. This allows it to have its own namespace in redis by tweaking the key the logger uses during the test.

I think this is ready for review again, @albertisfu.

albertisfu commented on cl/users/signals.py:
```diff
@@ -71,3 +79,35 @@ def complaint_handler(
+        UserProfile.objects.create(
+            user=instance,
+            activation_key=sha1_activation_key(instance.username),
+            key_expires=now() + timedelta(days=5),
```
Testing this signal, I think it would be good to set `email_confirmed=True` otherwise the superuser won't be able to log in, at least in development since they won't receive the verification email.
```suggestion
            key_expires=now() + timedelta(days=5),
            email_confirmed=True,
```

albertisfu reviewed (commented):
Wow, many new factories and test refactor! Everything looks good to me, just a small comment about the `superuser_creation` signal.

mlissner commented:
Cool. Thanks Alberto. Merging. Man, I regret doing this right now instead of other priorities, but I'm glad it's done.
